omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/DiagnosisResource2_2.java

```diff
@@ -0,0 +1,217 @@
+	 */
+	@Override
+	public Diagnosis save(Diagnosis diagnosis) {
+		return Context.getDiagnosisService().save(diagnosis);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#purge(java.lang.Object,
+	 *      org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public void purge(Diagnosis diagnosis, RequestContext requestContext) throws ResponseException {
+		Context.getDiagnosisService().purgeDiagnosis(diagnosis);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#getRepresentationDescription(org.openmrs.module.webservices.rest.web.representation.Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+		if (representation instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("diagnosis");
+			description.addProperty("patient", Representation.REF);
+			description.addProperty("condition");
+			description.addProperty("encounter");
+			description.addProperty("certainty");
+			description.addProperty("rank");
+			description.addProperty("voided");
+			description.addProperty("display");
+			description.addSelfLink();
+			return description;
+		} else if (representation instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("diagnosis");
+			description.addProperty("patient", Representation.REF);
+			description.addProperty("condition");
+			description.addProperty("encounter");
+			description.addProperty("certainty");
+			description.addProperty("rank");
+			description.addProperty("voided");
+			description.addProperty("auditInfo");
+			description.addProperty("display");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+	
+	@Override
+	public Model getGETModel(Representation rep) {
+		ModelImpl model = (ModelImpl) super.getGETModel(rep);
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			model
+			        .property("uuid", new StringProperty())
+			        .property("diagnosis", new StringProperty())
+			        .property("condition", new StringProperty())
+			        .property("certainty", new EnumProperty(ConditionVerificationStatus.class))
+			        .property("rank", new IntegerProperty())
+			        .property("patient", new RefProperty("#/definitions/PatientGetRef"))
+			        .property("voided", new BooleanProperty());
+		}
+		return model;
+	}
+	
+	/**
+	 * @param diagnosis
+	 * @return Diagnosis's name
+	 */
+	@PropertyGetter("display")
+	public String getDisplayString(Diagnosis diagnosis) {
+		if (diagnosis.getDiagnosis() == null) {
+			return "";
+		} else {
+			if (diagnosis.getDiagnosis().getCoded() != null)
```

Comment:
Can you use curly braces? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-CodeStyle